Sources/bip32-ed25519-swift/bip32-ed25519-swift.swift

```diff
@@ -227,14 +231,16 @@ public class Bip32Ed25519 {
         let zr = z.subdata(in: ED25519_SCALAR_SIZE ..< 2 * ED25519_SCALAR_SIZE)
 
         // left = kl + 8 * trunc28(zl)
-        // right = zr + kr
-
+        // right = zr + kr mod 2^256
         let left = BigUInt(Data(kl.reversed())) + BigUInt(Data(trunc256MinusGBits(zl: zl, g: g).reversed())) * BigUInt(8)
+        guard left < (BigUInt(1) << 255) else {
+            throw BigIntException.overflow
+        }
         let right = BigUInt(Data(kr.reversed())) + BigUInt(Data(zr.reversed()))
 
         // Reverse byte order back after calculations
         var leftData = Data(left.serialize().reversed())
-        var rightData = Data(right.serialize().reversed())
+        var rightData = Data(right.serialize().reversed()).prefix(ED25519_SCALAR_SIZE)
```

Comment:
Do we to prefix with 32bytes? What does it do?

Reply:
prefix just takes a slice of the specified length